connectors/src/lib/models/intercom.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  CreationOptional,
+  DataTypes,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+  Model,
+} from "sequelize";
+
+import { Connector, sequelize_conn } from "@connectors/lib/models";
+
+export class IntercomConnectorState extends Model<
+  InferAttributes<IntercomConnectorState>,
+  InferCreationAttributes<IntercomConnectorState>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare lastGarbageCollectionFinishTime?: Date;
+
+  declare connectorId: ForeignKey<Connector["id"]>;
+}
+
+IntercomConnectorState.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    lastGarbageCollectionFinishTime: {
+      type: DataTypes.DATE,
+      allowNull: true,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    modelName: "intercom_connector_states",
+    indexes: [{ fields: ["connectorId"], unique: true }],
+  }
+);
+
+Connector.hasOne(IntercomConnectorState);
+
+export class IntercomCollection extends Model<
+  InferAttributes<IntercomCollection>,
+  InferCreationAttributes<IntercomCollection>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare workspaceId: string;
+  declare collectionId: string;
+
+  declare helpCenterId: string | null;
+  declare parentId: string | null;
+
+  declare name: string;
+  declare description: string | null;
+
+  declare connectorId: ForeignKey<Connector["id"]>;
+}
+
+IntercomCollection.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    workspaceId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    collectionId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    helpCenterId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    parentId: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    description: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    indexes: [
+      { fields: ["workspaceId", "collectionId", "connectorId"], unique: true },
+      { fields: ["connectorId"] },
+      { fields: ["collectionId"] },
+    ],
+    modelName: "intercom_collections",
+  }
+);
+Connector.hasMany(IntercomCollection);
```

Comment:
do we allow null here?

Reply:
Yes, it means that a Connector instance can be associated with none to multiple IntercomCollection instances. TBH, I simply reused what we do for all others connectors 😄  

<img width="509" alt="Screenshot 2023-11-17 at 14 24 10" src="https://github.com/dust-tt/dust/assets/3803406/8b1e8608-f10c-4a43-8f8b-e7a0252922b7">

Reply:
yes you cargo culted my and Aric's broken code 😬

Reply:
it means that connector_id can be null right ?